Lesson-9/SenderApp/SenderApp/Controllers/AccountController.cs

```diff
@@ -0,0 +1,214 @@
+        private readonly IHttpContextAccessor _contextAccessor;
+
+        public AccountController(
+            SignInManager<CustomIdentityUser> signInManager,
+            IBussinessLayer bussinessLayer,
+            ILogger<AccountController> logger,
+            IMapper mapper,
+            UserManager<CustomIdentityUser> userManager,
+            IGateway gateway,
+            IReportService reportService,
+            ITemplateEngineService templateEngineService,
+            IHttpContextAccessor contextAccessor)
+        {
+            _signInManager = signInManager;
+            _bussinessLayer = bussinessLayer;
+            _logger = logger;
+            _mapper = mapper;
+            _userManager = userManager;
+            _gateway = gateway;
+            _reportService = reportService;
+            _templateEngineService = templateEngineService;
+            _contextAccessor = contextAccessor;
+        }
+        
+        [HttpGet]
+        public IActionResult Register()
+        {
+            return View();
+        }
+        
+        [HttpPost]
+        public async Task<IActionResult> Register(RegisterViewModel viewModel)
+        {
+            _logger.LogInformation($"Register start - {DateTime.Now}");
+            var actionName = ControllerContext.ActionDescriptor.ActionName;
+
+            if (ModelState.IsValid)
+            {
+                var requestToFacade = _mapper.Map<RequestFacadeDto>(viewModel);
+                var responseFromFacade = await _bussinessLayer.RegisterFacadeAsync(actionName, requestToFacade);
+                
+                if (responseFromFacade != null)
+                {
+                    // генерация токена для пользователя
+                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(responseFromFacade);
+                    
+                    var callbackUrl = Url.Action(
+                        "ConfirmEmail", 
+                        "Account", 
+                        new { userId = responseFromFacade.Id, code = code }, protocol: 
+                        HttpContext.Request.Scheme
+                        );
+
+                    //Устанавливаем импровизированный гейт Email
+                    await _gateway.SetGateAsync(new EmailService());
+                    
+                    var userAuthAttrib = new UserMessageDto()
+                    {
+                        Email = responseFromFacade.Email,
+                        CallBackUrl = callbackUrl,
+                    };
+                    
+                    //Прогоняем через шаблонизатор сообщение о подтверждении почты
+                    var message = await _templateEngineService.TemplateEngine(
+                        "UserAuthTemplate", 
+                        "userauthtemplate",
+                        null,
+                        userAuthAttrib);
+
+                    var currentUserName = _contextAccessor.HttpContext.User.FindFirst(ClaimTypes.Name)?.Value;

```

Comment:
```suggestion
                    var currentUserName = this.User.FindFirst(ClaimTypes.Name)?.Value;
```
вы можете обращаться к HttpContext или к юзеру напрямую из методов контроллера